cmake/cosmaConfig.cmake.in

```diff
@@ -12,49 +12,36 @@ if(NOT TARGET cosma::cosma)
 
     set(MKL_ROOT "@MKL_ROOT@")
     set(MPI_CXX_COMPILER "@MPI_CXX_COMPILER@")
-
-    find_package(MPI)
+    set(MPI_DETERMINE_LIBRARY_VERSION TRUE)
+    find_package(MPI COMPONENTS "CXX" REQUIRED)
 
     set(COSMA_BLAS "@COSMA_BLAS@")
     set(COSMA_SCALAPACK "@COSMA_SCALAPACK@")
-
-    if (${COSMA_BLAS} STREQUAL "MKL")
-        find_package(MKL)
-    elseif (${COSMA_BLAS} STREQUAL "CUDA" OR ${COSMA_BLAS} STREQUAL "ROCM")
-        set(TILEMM_GPU_BACKEND ${COSMA_BLAS} CACHE STRING FORCE "")
-        find_dependency(Tiled-MM) # bundled
-        if ("@COSMA_WITH_NCCL@")
-            if (${COSMA_BLAS} STREQUAL "CUDA")
-                find_package(CUDAToolkit)
-                find_package(NCCL)
-            else()
-                find_package(hip)
-                find_package(RCCL)
-            endif()
-        endif()
-        if ("@COSMA_WITH_GPU_AWARE_MPI@")
-            if (${COSMA_BLAS} STREQUAL "CUDA")
-                find_package(CUDAToolkit)
-            else()
-                find_package(hip)
-            endif()
+    set(COSMA_BLAS_VENDOR "@COSMA_BLAS_VENDOR@")
+    set(COSMA_BLAS_THREADING "@COSMA_BLAS_THREADING@")
+
+    if (${COSMA_BLAS} STREQUAL "CUDA" OR ${COSMA_BLAS} STREQUAL "ROCM")
+      if (${COSMA_BLAS} STREQUAL "CUDA")
+        find_package(CUDAToolkit)
+      else()
+        find_package(hip)
+      endif()
+
+      set(TILEMM_GPU_BACKEND ${COSMA_BLAS} CACHE STRING FORCE "")
+      find_dependency(Tiled-MM) # bundled
+
+      if ("@COSMA_WITH_NCCL@")
+        if (${COSMA_BLAS} STREQUAL "CUDA")
+          find_package(NCCL)
+        else()
+          find_package(RCCL)
         endif()
-    elseif (${COSMA_BLAS} STREQUAL "OPENBLAS")
-        find_package(OpenBLAS)
-        set_target_properties(openblas PROPERTIES 
-            INTERFACE_INCLUDE_DIRECTORIES "${OpenBLAS_INCLUDE_DIRS}"
-            IMPORTED_LOCATION "${OpenBLAS_LIBRARIES}")
-    elseif (${COSMA_BLAS} STREQUAL "CRAY_LIBSCI")
-        find_package(CRAY_LIBSCI REQUIRED)
-    elseif (${COSMA_BLAS} STREQUAL "CUSTOM")
-        find_package(BLAS)
+      endif()
+    else()
+        find_package(Blas)
     endif ()
 
-    if (${COSMA_SCALAPACK} STREQUAL "MKL")
-        find_package(MKL)
-    elseif (${COSMA_SCALAPACK} STREQUAL "CRAY_LIBSCI")
-        find_package(CRAY_LIBSCI REQUIRED)
-    elseif (${COSMA_SCALAPACK} STREQUAL "CUSTOM")
+    if (NOT ${COSMA_SCALAPACK} STREQUAL "NONE")
```

Comment:
Should we use "NONE" or "OFF" in this case for COSMA_SCALAPACK? In `miniapp/CMakeLists.txt`, there is `if(NOT COSMA_SCALAPACK MATCHES "OFF")`. 

Also, should we allow 'NONE' for COSMA_BLAS at all?

Reply:
I don't have a preference for either using `NONE` or `OFF` to indicate scalapack is not used.

Reply:
indeed OFF is better than NONE.